src/java.base/linux/classes/jdk/internal/platform/CgroupSubsystemFactory.java

```diff
@@ -46,6 +48,37 @@
     private static final String BLKIO_CTRL = "blkio";
     private static final String MEMORY_CTRL = "memory";
 
+    /*
+     * From https://www.kernel.org/doc/Documentation/filesystems/proc.txt
+     *
+     *  36 35 98:0 /mnt1 /mnt2 rw,noatime master:1 - ext3 /dev/root rw,errors=continue
+     *  (1)(2)(3)   (4)   (5)      (6)      (7)   (8) (9)   (10)         (11)
+     *
+     *  (1) mount ID:  unique identifier of the mount (may be reused after umount)
+     *  (2) parent ID:  ID of parent (or of self for the top of the mount tree)
+     *  (3) major:minor:  value of st_dev for files on filesystem
+     *  (4) root:  root of the mount within the filesystem
+     *  (5) mount point:  mount point relative to the process's root
+     *  (6) mount options:  per mount options
+     *  (7) optional fields:  zero or more fields of the form "tag[:value]"
+     *  (8) separator:  marks the end of the optional fields
+     *  (9) filesystem type:  name of filesystem of the form "type[.subtype]"
+     *  (10) mount source:  filesystem specific information or "none"
+     *  (11) super options:  per super block options
+     *
+     *
+     *  Pattern explanation:
+     *
+     *  /`````\    /`````\    /`````\    /```````\    /```````\    /```\ (8) /```````\     (10) + (11)
```

Comment:
Alternatively, consider the inline comment:

    private static final Pattern MOUNTINFO_PATTERN = Pattern.compile(
        "^[^\\s]+\\s+[^\\s]+\\s+[^\\s]+\\s+" +  // (1) + (2) + (3)
        "([^\\s]+)\\s+([^\\s]+)\\s+"         +  // (4) + (5) - group 1,2: mount point, mount options
        "[^-]+-\\s+"    +                       // (6) + (7) + (8)
        "([^\\s]+)\\s+" +                       // (9) - group 3: fstype
        ".*$");                                 // (10) + (11)

Reply:
Good suggestion. I've changed it to what you've suggested and removed the unused groups 1 and 2.